src/lib/components/QRCode.svelte

```diff
@@ -0,0 +1,115 @@
+<script lang="ts">
+  /**
+   * Inspired by:
+   *
+   * - Shoelace: https://github.com/shoelace-style/shoelace/blob/next/src/components/qr-code/qr-code.ts
+   * - DeckDeckGo: https://github.com/deckgo/deckdeckgo/blob/main/webcomponents/elements/src/components/qrcode/qrcode/qrcode.tsx
+   */
+  import { afterUpdate, createEventDispatcher } from "svelte";
+  import QrCreator from "qr-creator";
+  import { debounce } from "$lib/utils/debounce.utils";
+
+  export let ariaLabel: string | undefined = undefined;
+  export let value: string;
+
+  // Valid CSS colors
+  export let fillColor = "black";
+  export let backgroundColor = "white";
+
+  export let radius = 0;
+  // https://www.qrcode.com/en/about/error_correction.html
+  export let ecLevel: "L" | "M" | "Q" | "H" = "H";
+
+  let label: string;
+  $: label =
+    ariaLabel !== undefined && ariaLabel.length > 0 ? ariaLabel : value;
+
+  let container: HTMLDivElement | undefined;
+  let size: { width: number } | undefined = undefined;
+
+  // Add a small debounce in case the screen is resized
+  const initSize = debounce(() => {
+    if (container === undefined || container === null) {
+      size = undefined;
+      return;
+    }
+
+    const { width } = container.getBoundingClientRect();
+    size = {
+      width,
+    };
+  }, 25);
+
+  let once = false;
+  afterUpdate(() => {
+    if (once) {
+      return;
+    }
+
+    initSize();
+    once = true;
+  });
+
+  const dispatch = createEventDispatcher();
+
+  const renderCanvas = () => {
+    if (canvas === undefined || size === undefined) {
+      return;
+    }
+
+    QrCreator.render(
```

Comment:
Stupid question: How does QrCreator know that it must leave space in the center for the logo?

Reply:
There is no space. The logo is opaque and above the QR code.

Reply:
And that doesn't bother when you scan the QR code?

Reply:
Yes and no. The QR code contains a repetition of the code (x times the same code) so technically, you can display a logo because the code is repeated but, if the bigger the logo the more difficult for the QR code reader to read the logo. I warned the designers about it.

Reply:
Ok, thanks for the explanation. I'm not an expert on QRs.